openedx/features/course_card/views.py

```diff
@@ -42,19 +44,30 @@ def get_course_cards(request):
     current_time = datetime.utcnow()
 
     filtered_courses = []
+    programs = get_programs(request.site)
 
     for course in courses_list:
 
         if course.invitation_only and not CourseEnrollment.is_enrolled(request.user, course.id):
             continue
 
+        # check if any active (currently open or are starting in future)
+        # run of a course is part of a program.
+        course_program = [program for program in programs
```

Comment:
Let's please not use list comprehension if it affects readability or where ever we need to apply it at some nested levels, I tried hard but didn't get this logic, So I would request to open it up, make it readable and not use nested comprehensions.

Reply:
Second point is move it to a separate function and then divide that function to multiple if we can.